src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+                          // channelBinder,
+                          sessionToken, listenerSecurityState, settingsLifetimeManager, address)
+                {
+                    this.incomingChannel = (IReplyChannel) channel;
+                    base.OpenAsync(ServiceDefaults.OpenTimeout);
+                }
+
+                public IReplyChannel IncomingChannel
+                {
+                    get { return incomingChannel; }
+                    set { incomingChannel = value; }
+                }
+                protected override bool CanDoSecurityCorrelation
+                {
+                    get
+                    {
+                        return true;
+                    }
+                }
+
+                public Task DispatchAsync(RequestContext context)
+                {
+                    RequestContext securityRequestContext = base.ReceiveRequest(context);
+                    IServiceChannelDispatcher serviceChannelDispatcher = Settings.SecurityServiceDispatcher.
+                          GetInnerServiceChannelDispatcher(this.IncomingChannel);
+                   return serviceChannelDispatcher.DispatchAsync(securityRequestContext);
+                }
+
+                public Task DispatchAsync(Message message)
+                {
+                    throw new NotImplementedException();
+                }
+            }
+        }
+        class SecuritySessionRequestContext : RequestContextBase
+        {
+            RequestContext requestContext;
+            ServerSecuritySessionChannel channel;
+            SecurityProtocolCorrelationState correlationState;
+
+            public SecuritySessionRequestContext(RequestContext requestContext, Message requestMessage, SecurityProtocolCorrelationState correlationState, ServerSecuritySessionChannel channel)
+                : base(requestMessage, ServiceDefaults.CloseTimeout, ServiceDefaults.SendTimeout)
+            {
+                this.requestContext = requestContext;
+                this.correlationState = correlationState;
+                this.channel = channel;
+            }
+
+
+
+            protected override void OnAbort()
+            {
+                this.requestContext.Abort();
+            }
+
+            protected override Task OnCloseAsync(CancellationToken token)
+            {
+                return this.requestContext.CloseAsync(token);
+            }
+
+            protected override Task OnReplyAsync(Message message, CancellationToken token)
+            {
+                if (message != null)
+                {
+                    this.channel.SecureApplicationMessage(ref message, token, correlationState);
+                    //this.securityProtocol.SecureOutgoingMessageAsync(message);
+                    return this.requestContext.ReplyAsync(message);
+                }
+                else
+                {
+                    return Task.CompletedTask;
+                }
+            }
+
```

Comment:
Oprah Winfrey style blank lines here. You get a blank line, and you get a blank line, everyone gets a blank line!!

Reply:
'Oprah Winfrey' :)